java/arcs/core/entity/VariableEntityBase.kt

```diff
@@ -0,0 +1,41 @@
+package arcs.core.entity
+
+import arcs.core.common.Referencable
+import arcs.core.data.FieldName
+import arcs.core.data.RawEntity
+import arcs.core.data.Schema
+
+open class VariableEntityBase(
+    entityClassName: String,
+    schema: Schema,
+    entityId: String? = null,
+    creationTimestamp: Long = RawEntity.UNINITIALIZED_TIMESTAMP,
```

Comment:
should we leave the default value initialization to the superclass? Otherwise we have the default value defined twice and they could diverge, which would be bad.

Reply:
To impl this, can I just omit having default values here?

Reply:
I'm getting a build issue by removing this (admittedly, naively):

```
javatests/arcs/core/entity/DummyVariableEntity.kt:8:73: error: no value passed for parameter 'entityId'
class DummyVariableEntity : VariableEntityBase(ENTITY_CLASS_NAME, SCHEMA), Storable {
                                                                        ^
javatests/arcs/core/entity/DummyVariableEntity.kt:8:73: error: no value passed for parameter 'creationTimestamp'
class DummyVariableEntity : VariableEntityBase(ENTITY_CLASS_NAME, SCHEMA), Storable {
                                                                        ^
javatests/arcs/core/entity/DummyVariableEntity.kt:8:73: error: no value passed for parameter 'expirationTimestamp'
class DummyVariableEntity : VariableEntityBase(ENTITY_CLASS_NAME, SCHEMA), Storable {

```

Do you have a recommendation for what to do?

Reply:
I'm going to implement a default option, which is to revert this change.

Reply:
You can try changing these to arguments with default values being null, then null will be changed to default one layer down.

Reply:
I don't think that will work. Also, an idea comes to mind: I'll overload the ctor.

Reply:
ok, then maybe keeping the default here as well is ok.